### DISCUSSION TITLE
Inventory Materials in Project

### DISCUSSION THREAD

**Discussion by weizhongzou** [C-Accel-CRIPT/Python-SDK#179] [Q&A]
Does inventory also need to be contained in a project?

**Comment by nh916** [comment]
I believe inventory materials need to be contained in the project. 
@InnocentBug put in place checks that would try to look through the JSON and check for any orphaned nodes and be sure they are added to all the needed places, I don’t know too much about this. At some point we may need to revisit that to beef it up and check with the data model and try to figure out what nodes must be repeated, should be, and repetition is optional, to enforce good user inputs and guard against bad data input. 

For right now, if you are getting CRIPTOrphanedNodeError, then look more into it to see if it is right and fix the Python code, or if it is wrong and must change let us know.

**Reply by InnocentBug** [reply]
Yes, if you are getting `CRIPTOrphanedMaterialError` is is likely, that your material is not found on the inventories of the project, nor in the `project.material` list.

**Comment by InnocentBug** [comment]
I have to admit, that I don't fully understand the question.

What do you mean, with inventories have to be contained within a project?

If you are using an inventory (or more the materials from within the inventory) in a project, yes it has to be added to the project.

But keep in mind, that materials inside an inventory have to pre-exist somewhere else in CRIPT.
(So there has to be another project, that introduces this material.)

**Comment by InnocentBug** [comment]
If the question is, if you can have a stand-along Inventory and upload that into CRIPT.

The answer is no, you can't do that.

@brili set up the DB schema, such that materials have to exist somewhere else, and can only be referenced in Inventories.
So if you want to add an inventory, it makes sense to create a project, that introduces these materials and provides the inventory as well.
